Source/AwsCommonRuntimeKit/crt/Allocator.swift

```diff
@@ -40,11 +40,14 @@ internal extension Allocator {
     }
 
     /**
-     * Releases memory allocated by this allocator.
+     * Releases memory allocated by this allocator. We need to call deinitialize first so that Swift compiler's ARC
+     * can decreases the reference count of instances held by this pointer and release them if needed.
      *
      * - Parameter pointer: The pointer to allocated data.
+     * - Parameter count:
      */
-    func release<T>(_ pointer: UnsafeMutablePointer<T>?) {
```

Comment:
nit: setting default to 1 seems like an easy way to get into strange memory bugs down the road. as far as i can tell from docs count here is the number of elements that were allocated (ie same thing as capacity in allocate function), so its up to caller of this function to keep track of when to keep default and when to provide value and im guessing most of the time people will forget.